doc/guides/dts/design.rst

```diff
@@ -0,0 +1,92 @@
+.. _dt-design:
+
+Design goals
+############
+
+Zephyr's use of devicetree has evolved significantly over time, and further
+changes are expected. The following are the general design goals, along with
+specific examples about how they impact Zephyr's source code, and areas where
+more work remains to be done.
+
+Single source for all hardware information
+******************************************
+
+Zephyr shall obtain its hardware descriptions exclusively from devicetree.
+
+Examples
+========
+
+New device drivers shall use :ref:`existence macros <dt-existence-macros>` to
+determine whether a device is available and enabled.
+
+In-tree sample applications shall use :ref:`aliases <dt-alias-chosen>` to
+determine which of multiple possible generic devices of a given type will be
+used in the current build. For example, the :ref:`blinky-sample` uses this to
+determine the LED to blink.
+
+Example remaining work
+======================
+
+Zephyr's :ref:`sanitycheck_script` currently uses :file:`board.yaml` files to
```

Comment:
nit: imo this list should be bulleted :)

Reply:
I've made all the example lists bulleted. It looked weird to have some but not others bulleted when playing around with this. Also fixed a typo on this page.